src/components/pages/RequestAccessPage/RequestAccessPage.tsx

```diff
@@ -1,9 +1,227 @@
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAlertVisible, setIsAlertVisible] = useState(true);
+  const [isInfoCardVisible, setIsInfoCardVisible] = useState(true);
+  const [orgs, setOrgs] = useState<Org[]>([]);
+  const [accessRequests, setAccessRequests] = useState<AccessRequest[]>([]);
+
+  async function fetchOrgs() {
+    const response = await axios.get('api/org') as AxiosResponse<Org[]>;
+    setOrgs(response.data);
+  }
+
+  async function fetchAccessRequests() {
+    const response = await axios.get('api/user/access-requests') as AxiosResponse<AccessRequest[]>;
+    console.log('accessRequests', response.data);
+    setAccessRequests(response.data);
+  }
+
+  function handleEmailClick(org: Org) {
+    copyToClipboard(org.contact.email);
+  }
+
+  function handleRequestAccess(org: Org) {
+    async function requestAccess() {
+      const response = await axios.post(`api/access-request/${org.id}`) as AxiosResponse<AccessRequest>;
+      setAccessRequests([
+        ...accessRequests,
+        response.data,
+      ]);
+    }
+
+    requestAccess();
+  }
+
+  function copyToClipboard(text: string) {
+    const dummy = document.createElement('input');
+    document.body.appendChild(dummy);
+    dummy.setAttribute('value', text);
+    dummy.select();
+    document.execCommand('copy');
+    document.body.removeChild(dummy);
+  }
+
+  useEffect(() => {
+    async function fetchData() {
+      await Promise.all([
+        fetchOrgs(),
+        fetchAccessRequests(),
+      ]);
+
+      setIsLoading(false);
+    }
+
+    fetchData();
+  }, []);
+
+  if (isLoading) {
+    return <></>;
+  }
+
   return (
     <>
-      <h1>Request Access</h1>
+      <Container maxWidth="md">
+        <h1>My Groups</h1>
+
+        {/* Requests */}
+        {accessRequests.length > 0 && (
+          <Paper className={classes.paper}>
+            <TableContainer>
+              <Table size="small" aria-label="access requests table">
+                <TableHead>
+                  <TableRow>
+                    <TableCell>Group Name</TableCell>
+                    <TableCell>Contact</TableCell>
+                    <TableCell style={{ paddingLeft: '28px' }}>Email</TableCell>
+                    <TableCell>Phone</TableCell>
+                    <TableCell>Status</TableCell>
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {accessRequests.map(req => (
+                    <TableRow key={req.org.name}>
+                      <TableCell component="th" scope="row">
+                        {req.org.name}
+                      </TableCell>
+                      <TableCell>{`${req.org.contact.first_name} ${req.org.contact.last_name}`}</TableCell>
+                      <TableCell>
+                        <Tooltip title="Copy to clipboard" aria-label="copy email to clipboard">
+                          <IconButton onClick={() => handleEmailClick(req.org)} aria-label="email">
```

Comment:
Isn't it normal practice to use a mailto: link for emails?

Reply:
I was initially going to do that and then settled on copying the normal address to the clipboard for some reason. Maybe it's better to treat it like a regular email link though. I'll switch to that.